pushsync/pusher.go

```diff
@@ -202,16 +196,28 @@ func (p *Pusher) sync() {
 					unsubscribe()
 				}
 
-				// delete from pushed items
-				for i := 0; i < len(syncedAddrs); i++ {
-					delete(p.pushed, syncedAddrs[i].Hex())
-				}
-
 				// set chunk status to synced, insert to db GC index
 				if err := p.store.Set(ctx, chunk.ModeSetSyncPush, syncedAddrs...); err != nil {
 					log.Error("pushsync: error setting chunks to synced", "err", err)
 				}
 
+				// delete from pushed items
```

Comment:
Since we now increment synced tags in the localstore, we can only check if the root span is to be closed after we have performed the set operations.